src/rdkafka_cgrp.c

```diff
@@ -968,6 +979,25 @@ static void rd_kafka_cgrp_handle_JoinGroup (rd_kafka_t *rk,
                  * a matching assignor so error out early. */
                 ErrorCode = RD_KAFKA_RESP_ERR__BAD_MSG;
         }
+	else if (!ErrorCode) {
+                char *protocol_name = RD_KAFKAP_STR_DUP(&Protocol);
+                if (!(rkas = rd_kafka_assignor_find(rkcg->rkcg_rk,
+                                                    protocol_name)) ||
+                    !rkas->rkas_enabled) {
+                        rd_kafka_dbg(rkb->rkb_rk, CGRP, "JOINGROUP",
+                                "Unsupported assignment strategy \"%s\"",
+                                protocol_name);
+                        if (rkcg->rkcg_assignor) {
+                                if (rkcg->rkcg_assignor->rkas_destroy_state)
+                                        rkcg->rkcg_assignor->rkas_destroy_state(
+                                                rkcg->rkcg_assignor_state);
+                                rkcg->rkcg_assignor_state = NULL;
+                                rkcg->rkcg_assignor = NULL;
+                        }
+                        ErrorCode = RD_KAFKA_RESP_ERR__UNKNOWN_PROTOCOL;
```

Comment:
There is no explicit handling of this error code down in the err: label, what is the expected behaviour when this occurs?

Reply:
good question. i took the view that the behavior could be the same as if the protocol isn't set, which results in the error being `RD_KAFKA_RESP_ERR__BAD_MSG`, which is also not explicitly considered. what happens there, afaict, is back off `rkcg_join_intvl` then join the group again. perhaps not catastrophic enough, but i don't think that's too bad. opinion?

Reply:
The backoff and retry sounds like the right approach to support rolling upgrades of a group :+1: